zio-config/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -10,14 +10,14 @@ trait ConfigSource[K, V] {
 
 object ConfigSource {
   trait Service[K, V] {
-    def getConfigValue(path: List[K]): IO[ReadError[K, V], V]
+    def getConfigValue(path: Vector[K]): IO[ReadError[K, V], V]
```

Comment:
Addressing the PR review comment from @wi101 , on usage of Vector to ensure `:+` is effectively constant in time.